wallet/interface.go

```
@@ -0,0 +1,323 @@
+
+	// HaveAddress returns whether the wallet is the owner of the address.
+	HaveAddress(a btcutil.Address) (bool, error)
+
+	// ImportPublicKey imports a public key as a watch-only address.
+	ImportPublicKey(pubKey *btcec.PublicKey,
+		addrType waddrmgr.AddressType) error
+
+	// ImportTaprootScript imports a taproot script into the wallet.
+	ImportTaprootScript(scope waddrmgr.KeyScope,
+		tapscript *waddrmgr.Tapscript, bs *waddrmgr.BlockStamp,
+		witnessVersion byte, isSecretScript bool) (
+		waddrmgr.ManagedAddress, error)
+
+	// FetchDerivationInfo returns the derivation information for a given
+	// set of addresses.
+	FetchDerivationInfo(pkScript []byte) (*psbt.Bip32Derivation, error)
+
+	// CalculateBalance returns the wallet's total balance for a given
+	// number of required confirmations.
+	CalculateBalance(requiredConfirmations int32) (btcutil.Amount, error)
+
+	// CalculateAccountBalances returns the balances for a specific account.
+	CalculateAccountBalances(account uint32, requiredConfirmations int32) (
+		Balances, error)
+
+	// ListUnspent returns all unspent transaction outputs for a given
+	// account and confirmation requirement.
+	ListUnspent(minconf, maxconf int32, accountName string) (
+		[]*btcjson.ListUnspentResult, error)
+
+	// FetchOutpointInfo returns the output information for a given
+	// outpoint.
+	FetchOutpointInfo(prevOut *wire.OutPoint) (
+		*wire.MsgTx, *wire.TxOut, int64, error)
+
+	// LockOutpoint locks a specific UTXO, preventing it from being used in
+	// coin selection. The lock is identified by a unique ID and has an
+	// expiration time.
+	LockOutpoint(op wire.OutPoint)
+
+	// UnlockOutpoint unlocks a previously locked UTXO, making it available
+	// for coin selection again.
+	UnlockOutpoint(op wire.OutPoint)
+
+	// LockedOutpoint returns whether an outpoint has been marked as locked
+	// and should not be used as an input for created transactions.
+	LockedOutpoint(op wire.OutPoint) bool
```

I assume these operations interact with the database, right? If so, they can fail and should probably return an error

---

ok